src/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -50,6 +50,20 @@ internal static long TickCount64
             }
         }
 
+        public static string MachineName
+        {
+            get
+            {
+                const int MaxMachineNameLength = 256;
+                StringBuilder buf = new StringBuilder(MaxMachineNameLength);
+                int capacity = MaxMachineNameLength;
+                if (Interop.mincore.gethostname(buf, capacity) != 0)
```

Comment:
The other implementations of this properly for Windows call GetComputerName. Why are not we calling the same API here?